### DISCUSSION TITLE
Improving the item transfer when switching submarine

### DISCUSSION THREAD

**Discussion by AndrewDawnbringer** [FakeFishGames/Barotrauma#7974] [Ideas and suggestions]
Hello there!

- When purchasing and switching to a new sub, at first glance it looks like you've lost all your stuff. Thankfully, you did not, but you have it back on your previous submarine, so you need to transfer each item in your hands/toolbar/duffel/waist bag while switching back and forth which is kind of an unpleasant experience. 

- It will be nice to see a feature, when you switch sub, there is a popup, 'Would you like to transfer items from A submarine to B submarine?' , and if the player choose 'Yes', a Trade-like window pops up, where you see A submarine's inventory on the left, and B submarine's inventory on the right, and you can move the items on board via these panels. 

Or, it could be any other design, but I think you get it, some kind of tool which allow players to transfer items between submarines a fast and effective way.

_Best regards_

**Comment by Kafkatrap** [comment]
Yes, this needs to be a feature. 

Perhaps, when switching subs you get two additional windows, one with the contents of the first sub and another window showing a map of the 2nd sub with cabinets and storages marked.

Then you can drag and drop the items directly into the specified cabinets and storages on the new sub, so you don't have to spend hours sorting these items manually by putting them in your inventory first.

Something like that.

**Reply by glmsmoura** [reply]
That would be awesome as well! I like to organize things, but having tools to help me that is always welcomed.

**Comment by glmsmoura** [comment]
I think having some menu called "Stash Transfer" or something like this, where you can input items while you are on one sub (where you can input containers), and when you transfer to the other submarine, they appear as you bought them in the store, like @AndrewDawnbringer stated.

Just did my first sub transfer yesterday, and it was a huge pain.

**Reply by GregariousJB** [reply]
No offense, but a shared stash is an absolutely horrific idea. A transfer window or similar would be optimum. 

In order to handle the huge volume of items players would want to transfer between ships, a shared stash inventory would need to be massive to prevent needing to switch back and forth between subs over and over and over again just to take advantage of the shared stash. Now you have a massive storage box that people are abusing for normal storage. 

Even if the stash was small, and players didn't mind spending hours moving items from vanilla inventory to the stash, swapping subs, then moving items from the stash back to vanilla inventory on the new sub (repeated X times until everything is moved over), this means players that like to swap subs frequently for certain missions are needing to repeat this process each time they want to switch to another sub.

A well-made transfer window can't be abused and solves these issues for both player types - those that like to stick to one or two subs, and those that like to buy them all and switch freely between them.
